Stub CAM:U service

yuriks opened the pull request:
Also add valid stereo calbiration data. This lets Bravely Default get past the AR sequence and in-game.

Thanks to @mailwl for #1102 which was used for the stubs.

neobrain commented:
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-03T09:12:28Z: https://github.com/citra-emu/citra/compare/30fd5d9e9695758e7a10535a2709fa520f796aaa...yuriks:5e08ac8f9e4dc6665e0b52b85df59fc4213ed833

Subv commented on src/core/hle/service/cam/cam.cpp:
```diff
@@ -18,6 +18,256 @@
+    completion_event->Signal();
+
+    cmd_buff[1]    = RESULT_SUCCESS.raw;
+    cmd_buff[3]    = Kernel::g_handle_table.Create(completion_event).MoveFrom();
```
Same here

Subv commented on src/core/hle/service/cam/cam.cpp:
```diff
@@ -18,6 +18,256 @@
+    u8 port       = cmd_buff[1] & 0xFF;
+
+    cmd_buff[1]   = RESULT_SUCCESS.raw;
+    cmd_buff[3]   = Kernel::g_handle_table.Create(interrupt_error_event).MoveFrom();
```
Please add the handle descriptor to cmd_buff[2]

Subv commented on src/core/hle/service/cam/cam.cpp:
```diff
@@ -18,6 +18,256 @@
+    u8 port       = cmd_buff[1] & 0xFF;
+
+    cmd_buff[1]   = RESULT_SUCCESS.raw;
+    cmd_buff[3]   = Kernel::g_handle_table.Create(vsync_interrupt_error_event).MoveFrom();
```
Same here

Subv commented:
Please add some documentation to each of these functions

bunnei commented on src/core/hle/service/cam/cam.h:
```diff
@@ -141,6 +144,26 @@ enum class OutputFormat : u8 {
 
+/// Stereo camera calibration data.
+struct StereoCameraCalibrationData {
+    u8 isValidRotationXY;      ///< #bool Whether the X and Y rotation data is valid.
```
Is this a typo? (The #bool thing)

yuriks replied:
Was in the original commit. Seems to be a ctrulib thing denoting that the u8 is actually a boolean value. I can word it out.

neobrain commented:
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-13T09:00:54Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...yuriks:ec8a320cbe5d30787bce1e55d454e7afa2b2f7de

yuriks commented:
Updated. I added command headers and descriptors on all functions, and sorted them by id.

I haven't added documentation as requested by @Subv because I have no particular insight on most of these functions, any documentation I'd write would just be based on guesses from the function names.

neobrain commented:
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-13T09:14:05Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...yuriks:83d128624ea6364c0ddc702fc57386a296c779e8

neobrain commented:
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-13T09:16:36Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...yuriks:fc6fa0f088b3f536cc4333e66a34e082fc9a908b

Subv commented:
Just document the command inputs and outputs, like we do with all the other command handlers

yuriks commented:
Updated

Subv commented:
LGTM

bunnei commented:
LGTM
